### PR TITLE
Put "Demo" examples into the repo

### PR DESCRIPTION
**Pull request by warpech**
### Context

This PR adds "Demo" examples into the repo. The examples are based on:

- https://codesandbox.io/s/handsontable-javascript-data-grid-hello-world-app-1cc81?from-embed
- https://codesandbox.io/s/handsontable-typescript-data-grid-hello-world-app-tbsul?from-embed
- https://codesandbox.io/s/handsontable-angular-data-grid-hello-world-app-dhojw?from-embed
- https://codesandbox.io/s/handsontable-react-data-grid-hello-world-app-bhipr?from-embed
- https://codesandbox.io/s/handsontable-vue-data-grid-hello-world-app-5qgo7?from-embed
- @jansiegel's modifications to the above demos done in https://github.com/handsontable/handsontable/pull/8953

After this PR is merged to `develop` branch, it will be automatically deployed to:

- http://handsontable.github.io/handsontable/examples/11.1.0/docs/js/demo/index.html
- http://handsontable.github.io/handsontable/examples/11.1.0/docs/ts/demo/index.html
- http://handsontable.github.io/handsontable/examples/11.1.0/docs/angular/demo/index.html
- http://handsontable.github.io/handsontable/examples/11.1.0/docs/react/demo/index.html
- http://handsontable.github.io/handsontable/examples/11.1.0/docs/vue/demo/index.html
- http://handsontable.github.io/handsontable/examples/next/docs/js/demo/index.html
- http://handsontable.github.io/handsontable/examples/next/docs/ts/demo/index.html
- http://handsontable.github.io/handsontable/examples/next/docs/angular/demo/index.html
- http://handsontable.github.io/handsontable/examples/next/docs/react/demo/index.html
- http://handsontable.github.io/handsontable/examples/next/docs/vue/demo/index.html

### How has this been tested?

All the demos were verified locally by running `npm run build`, `npm run start` and then opening the example URL in the browser.

For JS and Angular, the template contains a test spec. It was also verified that `npm run test` works in these frameworks.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/8980 (first checkbox in the OP)
2. https://github.com/handsontable/handsontable/issues/8980#issuecomment-993697293 (points 1, 2, 4 from my comment)
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

File: examples/11.1.0/docs/angular/demo/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "ng": "ng",
+    "start": "ng serve --port 8080",
+    "build": "ng build",
+    "test": "node spec/support/jasmine.config.js"
+  },
+  "dependencies": {
+    "@angular/animations": "11.2.7",
+    "@angular/common": "11.2.7",
+    "@angular/compiler": "11.2.7",
+    "@angular/core": "11.2.7",
```
Comment (warpech) [inline examples/11.1.0/docs/angular/demo/package.json:24]: @jansiegel I have a question to you. After adding these new demos, none of the `package-lock.json` got modified. Is everything alright with `package-lock.json`? For example, in Angular examples, the `basic-example` and `demo` use different versions of Angular. Please advise if something is broken or should be immediately improved now.

Comment (jansiegel) [reply examples/11.1.0/docs/angular/demo/package.json:24]: The `package-lock.json` files inside `examples/[version]/docs/[framework]` directories are generated when running the `examples:install [version]` script from the root of the repo. What it does, is basically:
1. Clears any existing `package-lock` files and `node_modules` directories:
    ```
    Removing:
      11.1.0/**/(js|angular|react|vue)/node_modules
      11.1.0/**/(js|angular|react|vue)/**/node_modules
      11.1.0/**/(js|angular|react|vue)/**/dist
      11.1.0/**/(js|angular|react|vue)/**/.cache
      11.1.0/**/(js|angular|react|vue)/package-lock.json
    ```
    <sup>Excerpt from the log after calling `examples:install [version]`</sup>
2. Runs `npm i --no-audit` in all of the framework-based directories, so currently in:
    - `[version]/docs/angular`
    - `[version]/docs/react`
    - `[version]/docs/vue`
    - `[version]/docs/js`
3. Calls the `./scripts/link-packages.mjs` script to symlink some dependencies, if there's such a need (for examples, for the angular examples)

The `package-lock.json` files inside the example directories are a side-effect of running the examples manually, by calling `npm i` for each of the examples, so they serve another purpose.

I'm not sure if we should be committing any of these, to be honest, maybe it'd be a good idea, to add them to `.npmignore`.

**Review by jansiegel — CHANGES_REQUESTED** [review]
I have two notes:

1. The examples' helper scripts need to be updated, as they're missing the new `ts` section. Namely, the regexes in `examples/package.json` and `examples/scripts/clean-subpackages.mjs` come to mind. This is needed for the `examples:install` root script to work properly.
2. As previously all the "sections" of the examples were framework names (with the exception of `js`), there are places in the code where they're referred to as "frameworks", as in "framework-based directories" etc. With the addition of `ts`, I feel it's not as clear anymore. Maybe we should change those to something more accurate? WDYT?

**Comment by warpech** [comment]
Re 1. Thanks for poiting out! I copied your solution for `examples/package.json` and `examples/scripts/clean-subpackages.mjs` from https://github.com/handsontable/handsontable/pull/8953

@jansiegel, the PR https://github.com/handsontable/handsontable/pull/8953 also has changes in `examples/scripts/code-examples.mjs` and `examples/scripts/install-subpackages.mjs`. Should they be added here as well?

Re 2. Good catch, but I don't have a better word at the moment. Some people would say "flavours" or "variants", but I think that's even less informative.

**Comment by jansiegel** [comment]
> @jansiegel, the PR #8953 also has changes in `examples/scripts/code-examples.mjs` and `examples/scripts/install-subpackages.mjs`. Should they be added here as well?

@warpech Yup, they're mostly precautionary, but I think it'd be good to have them merged, thanks for noticing!

**Comment by warpech** [comment]
Done, with an exception - I didn't copy the improvement:

```
        // TODO: remove `--legacy-peer-deps` eventually
        await execa('npm', ['install', '--no-audit', '--legacy-peer-deps'], {
```

from `install-subpackages.mjs`, because that seems not needed anymore

**Review by jansiegel — APPROVED** [review]
:+1:

File: examples/11.1.0/docs/angular/demo/package.json (hunk 1 above)
Comment (warpech) [reply examples/11.1.0/docs/angular/demo/package.json:24]: OK, I get it. I added the single example's lockfiles to `.gitignore` and explained that in `README.md`. See 523a0ee71c0f0f9c118a0b618c6521a7f5e7298a. WDYT?

**Comment by warpech** [comment]
I additionally tested it by running:

- `npm run examples:install 11.1.0` - installed dependencies of `examples/next/docs/ts/demo` in `examples/next/docs/ts/node_modules`
- `npm run examples:version 12.0.0` - copied demos from `examples/next/docs/ts` to `examples/next/docs/ts` and seemingly added correct version information in relevant `package.json` files

**Pull request merged**